src/main/scala/org/broadinstitute/dsde/workbench/avram/util/AvramResult.scala

```diff
@@ -0,0 +1,20 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import cats.data.EitherT
+import cats.effect.IO
+
+// Note: I'm unsatisfied with AvramResult as a name but I can't think of anything better. Suggestions welcome. -breilly
+object AvramResult {
```

Comment:
what about EntityResult(s)?

Reply:
hmm, but I just realized it is also the return type from the queryAction call to sam. so maybe it does need to be more generic. maybe AvramDataOrError

Reply:
This is more general than that. It represents a potential result... or an error. It's a thing that you have to invoke to get that result. Until you do that, no actual computation has happened. It's essentially a program represented as a data structure that's built at runtime. It's really a specialization of cats `IO`... so something like `AvramIO` maybe.